exercises/danny_hindson/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,89 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include <algorithm>
+#include <cctype>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <string.h>
+#include <vector>
+
+bool compare(const std::pair<std::string,int>& x, const std::pair<std::string,int>& y) {
+  return x.second > y.second;
+}
+
+int main(int argc, char ** argv)
+{
+  std::string fileName;
+  if (argc == 2) {
+    fileName = argv[1];
+
+    std::string folderName ="D:\\newstarter\\builds\\WordCountDataFiles\\";
+    std::ifstream inFile(folderName + fileName);
+    
+    if (inFile) {
+      std::string s_withhyphens,s;
+      std::map<std::string, int> counters;
+
+      // read file in one word at a time
+      while (inFile >> s_withhyphens) {
+        std::istringstream ss(s_withhyphens);
+        // split any words containing hyphens into two
+        while (std::getline(ss, s, '-')) {
+          for (int i = 0; i < s.length(); i++) {
+            s[i] = tolower(s[i]);
+          }
+          char charsToRemove[] = ".,?'\"!():";
+          for (int i = 0; i < strlen(charsToRemove); i++) {
+            s.erase(std::remove(s.begin(), s.end(), charsToRemove[i]), s.end());
+          }
+          if (s.length() > 4) {
+            counters[s]++;
+          }
+        }
+      }
+
+      std::vector<std::pair<std::string, int>> vec;
```

Comment:
Can we avoid names like `vec`? They are a little generic and don't offer much meaning.